modules/apps/adaptive-media/adaptive-media-content-transformer-impl/src/main/java/com/liferay/adaptive/media/content/transformer/internal/HtmlContentTransformerImpl.java

```diff
@@ -39,42 +37,90 @@ public String transform(String html) throws PortalException {
 			return null;
 		}
 
-		if (!html.contains(AMImageHTMLConstants.ATTRIBUTE_NAME_FILE_ENTRY_ID)) {
-			return html;
-		}
+		StringBundler sb = new StringBundler();
 
-		return super.transform(html);
-	}
+		int i = 0;
 
-	@Override
-	protected FileEntry getFileEntry(Matcher matcher) throws PortalException {
-		long fileEntryId = GetterUtil.getLong(matcher.group(1));
+		while (i < html.length()) {
+			int imgStart = html.indexOf(_OPEN_TAG_TOKEN_IMG, i);
 
-		return _dlAppLocalService.getFileEntry(fileEntryId);
-	}
+			if (imgStart == -1) {
+				sb.append(html.substring(i));
 
-	@Override
-	protected Pattern getPattern() {
-		return _pattern;
-	}
+				break;
+			}
 
-	@Override
-	protected String getReplacement(String originalImgTag, FileEntry fileEntry)
-		throws PortalException {
+			int imgEnd = html.indexOf(
+				CharPool.GREATER_THAN, imgStart + _OPEN_TAG_TOKEN_IMG.length());
 
-		return _amImageHTMLTagFactory.create(originalImgTag, fileEntry);
-	}
+			if (imgEnd == -1) {
+				sb.append(html.substring(i));
 
-	@Override
-	protected boolean isSupported(FileEntry fileEntry) {
-		return _amImageMimeTypeProvider.isMimeTypeSupported(
-			fileEntry.getMimeType());
+				break;
+			}
+
+			imgEnd++;
+
+			int attributeStart = html.indexOf(
+				AMImageHTMLConstants.ATTRIBUTE_NAME_FILE_ENTRY_ID, imgStart);
+
+			if (attributeStart == -1) {
+				sb.append(html.substring(i));
+
+				break;
+			}
+
+			if (attributeStart > imgEnd) {
+				sb.append(html.substring(i, imgEnd));
+
+				i = imgEnd;
+
+				continue;
+			}
+
+			int fileEntryIdStart =
+				attributeStart +
+					AMImageHTMLConstants.ATTRIBUTE_NAME_FILE_ENTRY_ID.length() +
+						2;
+
+			int fileEntryIdEnd = html.indexOf(CharPool.QUOTE, fileEntryIdStart);
+
+			if ((fileEntryIdEnd == -1) || (fileEntryIdEnd > imgEnd)) {
+				sb.append(html.substring(i, imgEnd));
+
+				i = imgEnd;
+
+				continue;
+			}
+
+			long fileEntryId = GetterUtil.getLong(
+				html.substring(fileEntryIdStart, fileEntryIdEnd));
+
+			FileEntry fileEntry = _dlAppLocalService.getFileEntry(fileEntryId);
+
+			if (!_amImageMimeTypeProvider.isMimeTypeSupported(
+					fileEntry.getMimeType())) {
+
+				sb.append(html.substring(i, imgEnd));
+
+				i = imgEnd;
+
+				continue;
+			}
+
+			String replacement = _amImageHTMLTagFactory.create(
+				html.substring(imgStart, imgEnd), fileEntry);
+
+			sb.append(html.substring(i, imgStart));
+			sb.append(replacement);
+
+			i = imgEnd;
+		}
+
+		return sb.toString();
 	}
 
-	private static final Pattern _pattern = Pattern.compile(
-		"<img [^>]*?\\s*" + AMImageHTMLConstants.ATTRIBUTE_NAME_FILE_ENTRY_ID +
-			"=\"(\\d+)\".*?/?>",
-		Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
+	private static final String _OPEN_TAG_TOKEN_IMG = "<img ";
```

Comment:
Suggestion:

_OPEN_IMG_TAG seems more consistent with other portal existing patters as 

com.liferay.journal.content.web.internal.JournalContentPortletLayoutListener#_OPEN_TAG + com.liferay.commerce.product.internal.importer.CPFileImporterImpl#IMG_TAG

Reply:
I used this name because of this (related) rename: https://github.com/brianchandotcom/liferay-portal/pull/154031/commits/3ebdcd8a96f4a74b5dde728b1da4e2f9e23a0b94